RecoEgamma/EgammaIsolationAlgos/plugins/isolation_cones/PhotonPFIsolationWithMapBasedVeto.cc

```diff
@@ -0,0 +1,170 @@
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+
+#include "DataFormats/Candidate/interface/Candidate.h"
+#include "DataFormats/Candidate/interface/CandidateFwd.h"
+
+#include "DataFormats/EgammaCandidates/interface/Photon.h"
+#include "DataFormats/EgammaCandidates/interface/PhotonFwd.h"
+
+#include "DataFormats/PatCandidates/interface/Photon.h"
+
+#include "DataFormats/Common/interface/ValueMap.h"
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+
+#include "DataFormats/ParticleFlowCandidate/interface/PFCandidate.h"
+#include "DataFormats/PatCandidates/interface/PackedCandidate.h"
+
+#include "PhysicsTools/IsolationAlgos/interface/IsoDepositVetoFactory.h"
+#include "DataFormats/RecoCandidate/interface/IsoDepositVetos.h"
+
+#include "PhysicsTools/IsolationAlgos/interface/CITKIsolationConeDefinitionBase.h"
+#include "DataFormats/Math/interface/deltaR.h"
+
+#include "DataFormats/EcalDetId/interface/EcalSubdetector.h"
+
+
+#include <unordered_map>
+
+namespace reco {
+  typedef edm::Ptr<reco::Photon> recoPhotonPtr;
+}
+
+namespace pat {
+  typedef edm::Ptr<pat::Photon> patPhotonPtr;
+  typedef edm::Ptr<pat::PackedCandidate> PackedCandidatePtr;
+}
+
+// This template function finds whether theCandidate is in thefootprint
+// collection. It is templated to be able to handle both reco and pat
+// photons (from AOD and miniAOD, respectively).
+template <class T, class U>
```

Comment:
this and the template below should probably be in an anonymous namespace (wrap these templates in ``` namespace {  ... } ```  )